File: core/src/test/java/com/rudderstack/android/sdk/core/RudderLogger.java
package com.rudderstack.android.sdk.core;


/*
 * Logger class for RudderClient library
 * */
public class RudderLogger {
    private static int logLevel = RudderLogLevel.INFO;
    private static final String TAG = "RudderSDK";

    static void init(int l) {
        if (l > RudderLogLevel.VERBOSE) l = RudderLogLevel.VERBOSE;
        else if (l < RudderLogLevel.NONE) l = RudderLogLevel.NONE;
        logLevel = l;
    }

    public static void logError(Throwable throwable) {
        if (logLevel >= RudderLogLevel.ERROR) {
            System.out.println(TAG + " " + "Error: " + throwable);
        }
    }

    /*public  static void logError(Exception ex) {
        logError(ex.getMessage());
    }*/

    public static void logError(String message) {
        if (logLevel >= RudderLogLevel.ERROR) {
            System.out.println(TAG + " " + "Error: " + message);
        }
    }

    public static void logWarn(String message) {
        if (logLevel >= RudderLogLevel.WARN) {
            System.out.println(TAG + " "  + message);
        }
    }

    public static void logInfo(String message) {
        if (logLevel >= RudderLogLevel.INFO) {
            System.out.println(TAG + " " + message);
        }
    }

    public static void logDebug(String message) {
        if (logLevel >= RudderLogLevel.DEBUG) {
            System.out.println(TAG + " "+ message);
        }
    }

    public static void logVerbose(String message) {
        if (logLevel >= RudderLogLevel.VERBOSE) {
            System.out.println(TAG + " "+ message);
        }
    }

    public static class RudderLogLevel {
        public static final int VERBOSE = 5;
        public static final int DEBUG = 4;
        public static final int INFO = 3;
        public static final int WARN = 2;
        public static final int ERROR = 1;
        public static final int NONE = 0;
    }
}

File: core/src/main/java/com/rudderstack/core/RudderLogger.kt
/*
 * Creator: Debanjan Chatterjee on 02/02/22, 11:02 PM Last modified: 02/02/22, 11:02 PM
 * Copyright: All rights reserved Ⓒ 2022 http://rudderstack.com
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License. You may obtain a
 * copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

package com.rudderstack.core

/**
 * Logger interface.
 * Contains methods for different scenarios
 *
 */
interface RudderLogger {
    companion object{
        const val DEFAULT_TAG = "Rudder-Analytics"
    }

    /**
     * Activate or deactivate logger based on choice.
     *
     * @param level should log anything greater or equal to that level. See [LogLevel]
     */
    fun activate (level : LogLevel)

    fun info(tag : String = DEFAULT_TAG, log: String)

    fun debug(tag : String = DEFAULT_TAG, log: String)

    fun warn(tag : String = DEFAULT_TAG, log: String)

    fun error(tag : String = DEFAULT_TAG, log: String, throwable : Throwable? = null)

    /**
     * Level based on priority. Higher the number, greater the priority
     *
     * @property level priority for each type
     */
    enum class LogLevel{
        DEBUG,
        INFO,
        WARN,
        ERROR,
        NONE,
    }
    val level: LogLevel

    object Noob : RudderLogger{
        override fun activate(level: LogLevel) {
            // do nothing
        }

        override fun info(tag: String, log: String) {
            // do nothing
        }

        override fun debug(tag: String, log: String) {
            // do nothing
        }

        override fun warn(tag: String, log: String) {
            // do nothing
        }

        override fun error(tag: String, log: String, throwable: Throwable?) {
            // do nothing
        }

        override val level: LogLevel
            get() = LogLevel.NONE
    }
}
